### PR TITLE
[dmlib] Audit XML comments in `Azure.Storage.DataMovement` dmlib library

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR contains the XML comment changes for the `Azure.Storage.DataMovement` library.

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs [hunk 1]
```diff
@@ -191,7 +191,7 @@ public override Task<bool> TryRemoveStoredTransferAsync(string id, CancellationT
         /// <summary>
         /// Lists all the transfers contained in the checkpointer.
         /// </summary>
-        /// <param name="cancellationToken"></param>
+        /// <param name="cancellationToken">The cancellation token of the current transfer.</param>
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs:194]: Is it accurate to describe this as cancellation token of the current transfer? Or is it really the cancellation token of the checkpointer? Will change pending feedback

File: sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs [hunk 2]
```diff
@@ -20,12 +20,12 @@ internal abstract class JobPartInternal
         public QueueChunkDelegate QueueChunk { get; internal set; }
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs:20]: I realized this was internal so I stopped -- but leaving it here just to show what my changes look like.

File: sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs:194]: For everywhere that has cancellation token let's use
```
<param name="cancellationToken">
        /// Optional <see cref="CancellationToken"/> to propagate
        /// notifications that the operation should be cancelled.
        /// </param>
        ````

File: sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs [hunk 3]
```diff
@@ -96,15 +96,15 @@ public override Task TryAddTransferAsync(string id, CancellationToken cancellati
...
         /// </summary>
-        /// <param name="id"></param>
-        /// <param name="partNumber"></param>
-        /// <param name="offset"></param>
-        /// <param name="buffer"></param>
-        /// <param name="cancellationToken"></param>
+        /// <param name="id">The transfer ID.</param>
+        /// <param name="partNumber">The part number of the current transfer.</param>
+        /// <param name="offset">The offset of the current transfer as a long.</param>
+        /// <param name="buffer">The buffer of the current transfer in bytes.</param>
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs:106]: Maybe something like "The buffer to write data from to the checkpoint."

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.DataMovement/src/StorageTransferProgress.cs [hunk 4]
```diff
@@ -7,27 +7,27 @@
...
     {
         /// <summary>
-        /// Number of Files that succeeded in transfer
+        /// Number of files that were transferred successfully.
         /// </summary>
         public int CompletedCount { get; internal set; }
 
         /// <summary>
-        /// Number of files that were skipped in transfer due to overwrite being set to not overwrite files, but the files exists already
+        /// Number of files that were skipped in transfer due to files already existing in the destination and overwite policy set to not overwrite existing files.
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.DataMovement/src/StorageTransferProgress.cs:20]: Maybe we should rewrite this to be

Number of files that were skipped in transfer due to files already existing in the destination. Files will be skipped, if the overwrite policy is set to not overwrite existing files.

**Pull request merged**